pyvista/core/objects.py

```diff
@@ -489,26 +705,104 @@ def cube_map(self, flag):
+        deep : bool, optional
+            Perform a deep copy when ``True``. Shallow copy when ``False``.
+
         Returns
         -------
         pyvista.Texture
             Copied texture.
         """
-        return Texture(self.to_image().copy())
+        return Texture(self.to_image().copy(deep=deep))
 
     def to_skybox(self):
         """Return the texture as a ``vtkSkybox`` if cube mapping is enabled.
 
         Returns
         -------
         vtk.vtkSkybox
-            Skybox if cube mapping is enabled.  Otherwise, ``None``.
+            Skybox if cube mapping is enabled.  Otherwise, raises an exception.
+
+        Examples
+        --------
+        >>> from pyvista import examples
+        >>> texture = examples.download_sky_box_cube_map()  # doctest:+SKIP
+        >>> texture.to_skybox()  # doctest:+SKIP
+        <vtkmodules.vtkRenderingOpenGL2.vtkOpenGLSkybox(0x464dbb0) at 0x7f3130fab1c0>
 
         """
-        if self.cube_map:
-            skybox = _vtk.vtkSkybox()
-            skybox.SetTexture(self)
-            return skybox
+        if not self.cube_map:
+            raise ValueError('This texture is not a cube map.')
+        skybox = _vtk.vtkSkybox()
+        skybox.SetTexture(self)
+        return skybox
+
+    @property
+    def image_data(self):
+        """Return the underlying image data associated with this texture.
+
+        Returns
+        -------
+        pyvista.pyvista_ndarray
+            Array of the image data.
+
+        Examples
+        --------
+        >>> from pyvista import examples
+        >>> texture = examples.download_masonry_texture()
+        >>> texture.image_data.shape
+        (256, 256, 3)
+        >>> texture.image_data.dtype
+        dtype('uint8')
+
+        """
+        data = pyvista.wrap(self.GetInput().GetPointData().GetScalars())
+        if data.ndim > 1:
+            shape = (self.dimensions[1], self.dimensions[0], data.shape[1])
+        else:
+            shape = (self.dimensions[1], self.dimensions[0])
+
+        return np.flip(data.reshape(shape), axis=1).swapaxes(1, 0)
+
+    @image_data.setter
+    def image_data(self, image_data):
+        """Set the image data."""
+        self._from_array(image_data)
+
+    def to_greyscale(self):
+        """Convert this texture as a single component (greyscale) texture.
+
+        Returns
+        -------
+        pyvista.Texture
+            Texture converted to greyscale. If already black and white,
+
+        Notes
+        -----
+        The transparency channel (if available) will be dropped.
+
+        Uses NTSC/PAL implementation.
+
+        Examples
+        --------
+        >>> from pyvista import examples
+        >>> texture = examples.download_masonry_texture()
+        >>> bw_texture = texture.to_greyscale()
+        >>> bw_texture.plot()
+
+        """
+        if self.n_components == 1:
+            return self
+
+        new_texture = self.copy()
+        r, g, b = np.asarray(new_texture.image_data).T[:3]
```

Comment:
Why the `asarray` call? Isn't `new_texture.image_data` guaranteed to be a `pyvista_ndarray`?

Reply:
Correct, it's guaranteed to a pyvista_ndarray, but there's a memory leak (uncollected vtkweakref) if I don't "release" the original vtk array.

Reply:
I don't understand the mechanics here then. If the point is a copy, we should just call `.copy()` for clarity. If it's not about the copy, it's not clear to me how a reference sticks around. Isn't that something we should try to fix in, say, `pyvista_ndarray.__del__`?

Reply:
> If it's not about the copy, it's not clear to me how a reference sticks around

I agree that this is an issue. Would you be ok if I raised this as a separate issue?